.github/workflows/publish-npm.yml

```diff
@@ -0,0 +1,37 @@
+name: Publish release to npm
+
+on:
+  pull_request:
+    types: 
+      - closed
+    branches: 
+      - main
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    if: |
+      startsWith(github.head_ref, 'release-'')
+      && github.event.pull_request.merged
+      
+    steps: 
+      - name: Checkout out repository
+        uses: actions/checkout@v2
+
+      - name: Set up Node
+        uses: actions/setup-node@v2
+        with:
+          node-version: '12.x'
+          cache: 'yarn'
+      
+      # The yarn cache is not node_modules
+      - name: Install Dependencies
+        run: yarn --prefer-offline
+      
+      - name: Run tests
+        run: yarn test
+      
+      - name: Publish to npm
+        uses: JS-DevTools/npm-publish@v1
```

Comment:
We can't use this without implementing a strict key rotation policy. Updated docs to capture why, so that infra doesn't get the same question again in the future (hopefully)